query/src/main/java/org/infinispan/query/dsl/embedded/impl/QueryEngine.java

```diff
@@ -737,20 +737,27 @@ protected CacheQuery buildLuceneQuery(FilterParsingResult<?> filterParsingResult
       return query;
    }
 
-   protected Class<?> getTargetedClass(LuceneQueryParsingResult<?> parsingResult) {
+   protected Class<?> getTargetedClass(FilterParsingResult<?> parsingResult) {
       return (Class<?>) parsingResult.getTargetEntityMetadata();
    }
 
-   //todo [anistor] we could cache the result of the transformation if there are no parameters
-   protected LuceneQueryParsingResult<?> transform(FilterParsingResult<?> parsingResult, Map<String, Object> namedParameters) {
-      EntityNamesResolver entityNamesResolver = entityName -> {
-         try {
-            return Util.loadClassStrict(entityName, null);
-         } catch (ClassNotFoundException e) {
-            return null;
-         }
-      };
-      HibernateSearchPropertyHelper propertyHelper = new HibernateSearchPropertyHelper(getSearchFactory(), entityNamesResolver, null);
-      return JPALuceneTransformer.transform(parsingResult, getSearchFactory(), entityNamesResolver, propertyHelper::getDefaultFieldBridge, namedParameters);
+   private <TypeMetadata> LuceneQueryParsingResult<TypeMetadata> transform(FilterParsingResult<TypeMetadata> parsingResult, Map<String, Object> namedParameters) {
+      LuceneQueryParsingResult<TypeMetadata> luceneParsingResult;
+      if (queryCache != null && parsingResult.getParameterNames().isEmpty()) {
+         KeyValuePair<String, Class> queryCacheKey = new KeyValuePair<>(parsingResult.getJpaQuery(), LuceneQueryParsingResult.class);
+         luceneParsingResult = queryCache.get(queryCacheKey);
+         if (luceneParsingResult == null) {
+            luceneParsingResult = createLuceneMaker().transform(parsingResult, namedParameters, getTargetedClass(parsingResult));
+            queryCache.put(queryCacheKey, luceneParsingResult);
```

Comment:
Lucene queries are not serializable, we have externalizers for some of them, I suppose all the queries generated by ```createLuceneMaker``` are covered?